src/kayenta/layout/table/nativeTable.tsx

```diff
@@ -0,0 +1,50 @@
+import * as React from 'react';
+import * as classNames from 'classnames';
+
+import { ITableColumn } from './tableColumn';
+import { NativeTableHeader } from './nativeTableHeader';
+
+export interface INativeTableProps<T> {
+  rows: T[];
+  columns: ITableColumn<T>[];
+  rowKey: (row: T) => string;
+  tableBodyClassName?: string;
+  headerClassName?: string;
+  rowClassName?: (row: T) => string;
+  onRowClick?: (row: T) => void;
+  customRow?: (row: T) => JSX.Element;
+  className?: string;
+}
+
+export function NativeTable<T>({ rows, columns, rowKey, tableBodyClassName, rowClassName, onRowClick, customRow, className, headerClassName }: INativeTableProps<T>) {
```

Comment:
A note on this: the `Table` component uses flexbox in a way that makes it very intolerant of variance between rows, and to this point has avoided layout breakage due to only using very specific pieces of data that don't produce that much variance. For lists like the report list, it's arguably much more appropriate to use the browser's table layout engine to make sure we're presenting columns at the best width based on their contents and are all horizontally aligned across rows.

The original `Table` is very tied to its own expectations about flex layouts, and as a result is very brittle to move to table layout and/or make configurable. Given that we use `Table` for several things that are really lists, not tables, I've chosen here to just accept some mild duplication in the hopes of going back and moving the list-y uses of `Table` to some new component and all the table-y uses to this table component.

Reply:
Good points here, thanks.